lib/fast_jsonapi/serialization_core.rb

```diff
@@ -65,55 +64,102 @@ def meta_hash(record, params = {})
         FastJsonapi.call_proc(meta_to_serialize, record, params)
       end
 
+      def generate_cache_key(record, namespace)
+        # using ':' as delimiters
+        [
+          namespace,
+          record.cache_key
+        ].join(':')
+      end
+
       def record_hash(record, fieldset, includes_list, params = {})
         if cache_store_instance
-          cache_opts = record_cache_options(cache_store_options, fieldset, includes_list, params)
-          record_hash = cache_store_instance.fetch(record, **cache_opts) do
-            temp_hash = id_hash(id_from_record(record, params), record_type, true)
-            temp_hash[:attributes] = attributes_hash(record, fieldset, params) if attributes_to_serialize.present?
-            temp_hash[:relationships] = relationships_hash(record, cachable_relationships_to_serialize, fieldset, includes_list, params) if cachable_relationships_to_serialize.present?
-            temp_hash[:links] = links_hash(record, params) if data_links.present?
-            temp_hash
+          cache_opts = record_cache_options(record, cache_store_options, fieldset, includes_list, params)
+          name_space = cache_opts.delete(:namespace)
+          cache_key = generate_cache_key(record, name_space)
+          klass = self
+          param = {
```

Comment:
```suggestion
          fetch_query = {
```
`param` and `params` are a bit confusing